### ISSUE TITLE
Panic: invalid memory address or nil dereference

### ISSUE THREAD

**Issue by lilshady** [davidbyttow/govips#128]
libvips version: libvips 8.10.2-Mon Oct 12 16:43:59 UTC 2020
govips: v2.1.0
go version: go1.15.5 darwin/amd64
macOs: 10.15.6 

I wrote a web service which calls the govips to resize pictures like:
```
vips.Startup(&vips.Config{MaxCacheFiles: 0,MaxCacheMem: 0,MaxCacheSize: 0})

i, err := vips.NewImageFromFile(filename)
	err  = i.Resize(0.5, vips.KernelAuto)

	if err != nil {
		return
	}
	ep := vips.NewDefaultJPEGExportParams()
	_, _, err = i.Export(ep)
```
int the stress testing "wrk -t10 -c200 -d60s http://localhost:7777/image/test" , I found some errors like and it is reproducable: 
```
2020/12/03 19:03:00 http: panic serving [::1]:56984: runtime error: invalid memory address or nil pointer dereference
goroutine 1383 [running]:
net/http.(*conn).serve.func1(0xc0097b7180)
        /usr/local/go/src/net/http/server.go:1801 +0x147
panic(0x42ebaa0, 0x45d9420)
        /usr/local/go/src/runtime/panic.go:975 +0x47a
github.com/davidbyttow/govips/vips.(*ImageRef).PremultiplyAlpha(0x0, 0xb7c, 0x0)
        /Users/xxx/go/src/github.com/davidbyttow/govips/vips/image.go:397 +0x26
github.com/davidbyttow/govips/vips.(*ImageRef).Resize(0x0, 0x3fe0000000000000, 0xffffffffffffffff, 0x43a1be0, 0xc000012540)
        /Users/xxx/go/src/github.com/davidbyttow/govips/vips/image.go:681 +0x2f
main.testGOVips(0x43a75a0, 0xc0000e7a40, 0xc00e777500)
        /Users/xxx/go/src/test/main.go:130 +0x138
net/http.HandlerFunc.ServeHTTP(0x435d9e8, 0x43a75a0, 0xc0000e7a40, 0xc00e777500)
        /usr/local/go/src/net/http/server.go:2042 +0x44
net/http.(*ServeMux).ServeHTTP(0xc000070380, 0x43a75a0, 0xc0000e7a40, 0xc00e777500)
        /usr/local/go/src/net/http/server.go:2417 +0x1ad
net/http.serverHandler.ServeHTTP(0xc0001be000, 0x43a75a0, 0xc0000e7a40, 0xc00e777500)
        /usr/local/go/src/net/http/server.go:2843 +0xa3
net/http.(*conn).serve(0xc0097b7180, 0x43a7b60, 0xc00073c980)
        /usr/local/go/src/net/http/server.go:1925 +0x8ad
created by net/http.(*Server).Serve
        /usr/local/go/src/net/http/server.go:2969 +0x36c

```
Do you have any idea?
Thanks

**Comment by tonimelisma** [comment]
Thanks for this. Unfortunately debugging with just the stack trace is nearly impossible and I guess you can't get any more info by putting a debugger into the program either.

If I read correctly:

```github.com/davidbyttow/govips/vips.(*ImageRef).PremultiplyAlpha(0x0, 0xb7c, 0x0)```

```github.com/davidbyttow/govips/vips.(*ImageRef).Resize(0x0, 0x3fe0000000000000, 0xffffffffffffffff, 0x43a1be0, 0xc000012540)```

The first parameter for a Golang method is the associated struct. In both PremultiplyAlpha() as well as the parent Resize() the ImageRef seems to be 0x0 or nil in plain English. So you're trying to resize an uninitialized ImageRef.

I suggest checking the error from vips.NewImageFromFile() as likely that's already failing and not opening a picture. Your code doesn't check the error before calling Resize(). If you do so, and you get the error, please provide it to us in case it's some error related to govips.

**Comment by lilshady** [comment]
Thanks for your quick response. I have made the change as you suggested and  It really fails to open the image. I guess some test files are bad formatted. 
Thanks a lot for your help!
